Makefile

```diff
@@ -41,6 +42,42 @@ FILTER_ERRCHECK := grep -v $(patsubst %,-e %, $(ERRCHECK_EXCLUDES))
 CHANGELOG_VERSION = $(shell grep '^v[0-9]' CHANGELOG.md | head -n1 | cut -d' ' -f1)
 INTHECODE_VERSION = $(shell perl -ne '/^const Version.*"([^"]+)".*$$/ && print "v$$1\n"' version.go)
 
+CI_TYPES ?= lint test examples crossdock
+ifneq ($(filter lint,$(CI_TYPES)),)
+CI_LINT := true
+endif
+ifneq ($(filter test,$(CI_TYPES)),)
+CI_TEST := true
+endif
+ifneq ($(filter cover,$(CI_TYPES)),)
+CI_COVER := true
+endif
+ifneq ($(filter examples,$(CI_TYPES)),)
+CI_EXAMPLES := true
+endif
+ifneq ($(filter crossdock,$(CI_TYPES)),)
+CI_CROSSDOCK := true
+endif
+ifneq ($(filter goveralls,$(CI_TYPES)),)
+CI_GOVERALLS := true
+endif
+
+CI_CACHE_DIR := $(shell pwd)/.cache
+CI_DOCKER_CACHE_DIR := $(CI_CACHE_DIR)/docker
+CI_DOCKER_IMAGE := yarpc_go
+CI_DOCKER_CACHE_FILE := $(CI_DOCKER_CACHE_DIR)/$(CI_DOCKER_IMAGE)
+
+DOCKER_IMAGE := yarpc/yarpc-go:latest
+
+DOCKER_COMPOSE_VERSION := 1.10.0
+
+_BIN_DIR = $(CI_CACHE_DIR)/bin
+
+$(_BIN_DIR)/docker-compose:
+	mkdir -p $(_BIN_DIR)
+	curl -L https://github.com/docker/compose/releases/download/$(DOCKER_COMPOSE_VERSION)/docker-compose-$(shell uname -s)-$(shell uname -m) > $(_BIN_DIR)/docker-compose
+	chmod +x $(_BIN_DIR)/docker-compose
```

Comment:
Do we still need this? The Travis Docker environment comes with docker-compose; are we using features that are only available in this particular release?